File: test-plugins/org.yakindu.sct.generator.cpp.test/gtests/EntryExitSelfTransitionTest/EntryExitSelfTransitionTest.cc
/**
* Copyright (c) 2014 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include <string>
#include "gtest/gtest.h"
#include "EntryExitSelfTransition.h"

TEST(StatemachineTest, SelfTransitionToChildState) {
	EntryExitSelfTransition* statechart = new EntryExitSelfTransition();
	statechart->init();
	statechart->enter();
	statechart->runCycle();
	EXPECT_TRUE(statechart->getDefaultSCI()->get_entries()== 1);
	EXPECT_TRUE(statechart->isActive(EntryExitSelfTransition::main_region_A__region0_B));
	statechart->getDefaultSCI()->set_entries(0);
	statechart->raise_e();
	statechart->runCycle();
	EXPECT_TRUE(statechart->getDefaultSCI()->get_entries()== 1);
	EXPECT_TRUE(statechart->getDefaultSCI()->get_exits()== 1);
	EXPECT_TRUE(statechart->isActive(EntryExitSelfTransition::main_region_A__region0_C));
	delete statechart;
}
TEST(StatemachineTest, SelfTransitionFromChildState) {
	EntryExitSelfTransition* statechart = new EntryExitSelfTransition();
	statechart->init();
	statechart->enter();
	statechart->runCycle();
	EXPECT_TRUE(statechart->getDefaultSCI()->get_entries()== 1);
	statechart->getDefaultSCI()->set_entries(0);
	statechart->raise_e1();
	statechart->runCycle();
	EXPECT_TRUE(statechart->getDefaultSCI()->get_entries()== 0);
	EXPECT_TRUE(statechart->getDefaultSCI()->get_exits()== 0);
	EXPECT_TRUE(statechart->isActive(EntryExitSelfTransition::main_region_A__region0_C));
	statechart->raise_e1();
	statechart->runCycle();
	EXPECT_TRUE(statechart->isActive(EntryExitSelfTransition::main_region_A__region0_B));
	EXPECT_TRUE(statechart->getDefaultSCI()->get_entries()== 1);
	EXPECT_TRUE(statechart->getDefaultSCI()->get_exits()== 1);
	delete statechart;
}

File: test-plugins/org.yakindu.sct.generator.csharp.test/test-gen/org/yakindu/sct/generator/csharp/test/EntryExitSelfTransitionTest.java
/**
 * Copyright (c) 2015 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */

package org.yakindu.sct.generator.csharp.test;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.*;
import org.yakindu.scr.entryexitselftransition.EntryExitSelfTransitionStatemachine;
import org.yakindu.scr.entryexitselftransition.EntryExitSelfTransitionStatemachine.State;
/**
 *  Unit TestCase for EntryExitSelfTransition
 */
@SuppressWarnings("all")
public class EntryExitSelfTransitionTest {

	private EntryExitSelfTransitionStatemachine statemachine;

	@Before
	public void setUp() {
		statemachine = new EntryExitSelfTransitionStatemachine();
		statemachine.init();
	}

	@After
	public void tearDown() {
		statemachine = null;
	}

	@Test
	public void testSelfTransitionToChildState() {
		statemachine.enter();
		statemachine.runCycle();
		assertTrue(statemachine.getEntries() == 1l);
		assertTrue(statemachine.isStateActive(State.main_region_A__region0_B));
		statemachine.setEntries(0l);
		statemachine.raiseE();
		statemachine.runCycle();
		assertTrue(statemachine.getEntries() == 1l);
		assertTrue(statemachine.getExits() == 1l);
		assertTrue(statemachine.isStateActive(State.main_region_A__region0_C));
	}
	@Test
	public void testSelfTransitionFromChildState() {
		statemachine.enter();
		statemachine.runCycle();
		assertTrue(statemachine.getEntries() == 1l);
		statemachine.setEntries(0l);
		statemachine.raiseE1();
		statemachine.runCycle();
		assertTrue(statemachine.getEntries() == 0l);
		assertTrue(statemachine.getExits() == 0l);
		assertTrue(statemachine.isStateActive(State.main_region_A__region0_C));
		statemachine.raiseE1();
		statemachine.runCycle();
		assertTrue(statemachine.isStateActive(State.main_region_A__region0_B));
		assertTrue(statemachine.getEntries() == 1l);
		assertTrue(statemachine.getExits() == 1l);
	}
}
